### PR TITLE
feat(ELB): add elb l7rules data source

### PR DESCRIPTION
**Pull request by SabreKeyZ**
**What this PR does / why we need it**:
 add elb l7rules data source
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
 make testacc TEST=./huaweicloud/services/acceptance/elb/ TESTARGS='-run TestAccDatasourceL7rules_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/elb/ -v -run TestAccDatasourceL7rules_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceL7rules_basic
=== PAUSE TestAccDatasourceL7rules_basic
=== CONT  TestAccDatasourceL7rules_basic
--- PASS: TestAccDatasourceL7rules_basic (125.78s) 
PASS                                                                                                             
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/elb       125.822s 
```

### CODE REVIEW

File: docs/data-sources/elb_l7rules.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+---
+subcategory: Dedicated Load Balance (Dedicated ELB)
+---
+
+# huaweicloud_elb_l7rules
+
+Use this data source to get the list of ELB l7 rules.
```
Comment (houpeng80) [inline docs/data-sources/elb_l7rules.md]:
```suggestion
Use this data source to get the list of ELB L7 rules.
```

File: docs/data-sources/elb_l7rules.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+<a name="Elb_l7rules"></a>
+The `l7rules` block supports:
+
+* `id` - The ID of the forwarding policy.
+
+* `type` - The type of the forwarding rule.
+
+* `value` - The value of the match item.
+
+* `compare_type` - The requests are matched with the domain name or URL.
```
Comment (houpeng80) [inline docs/data-sources/elb_l7rules.md]:
```suggestion
* `compare_type` - How the requests are matched with the domain name or URL.
```

File: docs/data-sources/elb_l7rules.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+* `l7policy_id` - (Required, String) Specifies the forwarding policy ID.
+
+* `l7rule_id` - (Optional, String) Specifies the forwarding rule ID.
+
+* `compare_type` - (Optional, String) Specifies how requests are matched with the domain names or URL. Values options:
+  **EQUAL_TO**, **REGEX**, **STARTS_WITH**.
+
+* `value` - (Optional, String) Specifies the value of the match content.
+
+* `type` - (Optional, String) Specifies the match type.
```
Comment (houpeng80) [inline docs/data-sources/elb_l7rules.md]: add the value options refer to the resource of huaweicloud_elb_l7rule

File: huaweicloud/services/acceptance/elb/data_source_huaweicloud_elb_l7rules_test.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+					resource.TestCheckOutput("type_filter_is_useful", "true"),
+				),
+			},
+		},
+	})
+}
+
+func testAccDatasourceL7rules_basic(name string) string {
+	return fmt.Sprintf(`
+%[1]s
```
Comment (houpeng80) [inline huaweicloud/services/acceptance/elb/data_source_huaweicloud_elb_l7rules_test.go:44]: add a new line

File: huaweicloud/services/elb/data_source_huaweicloud_elb_l7rules.go [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	listL7rulesPath = strings.ReplaceAll(listL7rulesPath, "{l7policy_id}", d.Get("l7policy_id").(string))
+	listL7rulesQueryParams := buildListL7rulesQueryParams(d)
+	listL7rulesPath += listL7rulesQueryParams
+	listL7rulesResp, err := pagination.ListAllItems(
+		listL7rulesClient,
+		"marker",
+		listL7rulesPath,
+		&pagination.QueryOpts{MarkerField: ""})
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving ELB L7rules")
```
Comment (houpeng80) [inline huaweicloud/services/elb/data_source_huaweicloud_elb_l7rules.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "error retrieving ELB L7 rules")
```

Comment (SabreKeyZ) [reply huaweicloud/services/elb/data_source_huaweicloud_elb_l7rules.go]: done

File: huaweicloud/services/acceptance/elb/data_source_huaweicloud_elb_l7rules_test.go (hunk 4 above)
Comment (SabreKeyZ) [reply huaweicloud/services/acceptance/elb/data_source_huaweicloud_elb_l7rules_test.go:44]: done

File: docs/data-sources/elb_l7rules.md (hunk 2 above)
Comment (SabreKeyZ) [reply docs/data-sources/elb_l7rules.md]: done

File: docs/data-sources/elb_l7rules.md (hunk 1 above)
Comment (SabreKeyZ) [reply docs/data-sources/elb_l7rules.md]: done

File: docs/data-sources/elb_l7rules.md (hunk 3 above)
Comment (SabreKeyZ) [reply docs/data-sources/elb_l7rules.md]: done

**Pull request merged**